plugins/inputs/sql/sql.go

```diff
@@ -0,0 +1,809 @@
+			dest_type = TYPE_AUTO
+			cell = new(sql.RawBytes)
+		} else if !s.IgnoreOtherFields {
+			dest_type = TYPE_AUTO
+			cell = new(sql.RawBytes)
+		} else {
+			field_matched = false
+			cell = new(sql.RawBytes)
+			if Debug {
+				log.Printf("I! Skipped field %s", s.column_name[i])
+			}
+		}
+
+		if Debug && !field_matched {
+			log.Printf("I! Column %d '%s' dest type  %d", i, s.column_name[i], dest_type)
+		}
+
+		if s.column_name[i] == s.FieldHost {
+			s.field_host_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldDatabase {
+			s.field_database_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldMeasurement {
+			s.field_measurement_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldName {
+			s.field_name_idx = i
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldValue {
+			s.field_value_idx = i
+			s.field_value_type = dest_type
+			field_matched = false
+		}
+		if s.column_name[i] == s.FieldTimestamp {
+			s.field_timestamp_idx = i
+			field_matched = false
+		}
+
+		if field_matched {
+			s.field_type[s.field_count] = dest_type
+			s.field_idx[s.field_count] = i
+			s.field_count++
+		}
+		s.cells[i] = cell
+		s.cell_refs[i] = &s.cells[i]
+	}
+
+	if Debug {
+		log.Printf("I! Query structure with %d tags and %d fields on %d columns...", s.tag_count, s.field_count, col_count)
+	}
+
+	return nil
+}
+
+func ConvertString(name string, cell interface{}) (string, bool) {
+	value, ok := cell.(string)
+	if !ok {
+		var barr []byte
+		barr, ok = cell.([]byte)
+		if !ok {
+			var ivalue int64
+			ivalue, ok = cell.(int64)
+			if !ok {
+				value = fmt.Sprintf("%v", cell)
+				ok = true
+				if Debug {
+					log.Printf("W! converting '%s' type %s raw data '%s'", name, reflect.TypeOf(cell).Kind(), fmt.Sprintf("%v", cell))
+				}
+			} else {
+				value = strconv.FormatInt(ivalue, 10)
+			}
+		} else {
+			value = string(barr)
+		}
+	}
+	return value, ok
+}
+
+func (s *Query) ConvertField(name string, cell interface{}, field_type int, NullAsZero bool) (interface{}, error) {
+	var value interface{}
+	var ok bool
+	var str string
+	var err error
+
+	ok = true
+	if cell != nil {
+		switch field_type {
+		case TYPE_INT:
+			str, ok = cell.(string)
+			if ok {
+				value, err = strconv.ParseInt(str, 10, 64)
+			}
+			break
+		case TYPE_FLOAT:
+			str, ok = cell.(string)
+			if ok {
+				value, err = strconv.ParseFloat(str, 64)
+			}
+			break
+		case TYPE_BOOL:
+			str, ok = cell.(string)
+			if ok {
+				value, err = strconv.ParseBool(str)
+			}
+			break
+		case TYPE_TIME:
+			value, ok = cell.(time.Time)
+			if !ok {
+				var intvalue int64
+				intvalue, ok = value.(int64)
+				if ok {
+					// TODO convert to s/ms/us/ns??
+					value = time.Unix(intvalue, 0)
+				}
+			}
+			break
+		case TYPE_STRING:
+			value, ok = ConvertString(name, cell)
+			break
+		default:
+			value = cell
+		}
+	} else if NullAsZero {
+		switch field_type {
```

Comment:
again unnecessary breaks. you can also [batch up switch cases](https://play.golang.org/p/Jq13KG4w3j)